src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -132,8 +169,12 @@ public synchronized SpoolMessage addMessage(Publish request) throws InterruptedE
         return message;
     }
 
-    private void addMessageToSpooler(long id, SpoolMessage message) {
-        spooler.add(id, message);
+    private void addMessageToSpooler(long id, SpoolMessage message) throws SpoolerStoreException {
+        try {
+            spooler.add(id, message);
+        } catch (IOException e) {
+            throw new SpoolerStoreException(e);
```

Comment:
must fallback

Reply:
We don't necessarily want to switch the spooler, we just want to fallback. So when writing, try writing to disk then fallback if needed.
When reading, read from memory first, then read from disk.

Reply:
Yes, we need to check to see if there are any messages which were put their as the fallback, otherwise they'd never be sent which is very bad.